src/plugins/walletConnectToDapps/v2/eventsManager/useWalletConnectEventsManager.ts

```diff
@@ -38,17 +41,32 @@ export const useWalletConnectEventsManager = (
       const signClientEvents = web3wallet.engine.signClient.events
       const pairingEvents = core.pairing.events
 
-      const sessionRequestListener = (request: Web3WalletTypes.SessionRequest) =>
+      const sessionRequestListener = (
+        request: SignClientTypes.EventArguments['session_request'],
+      ) => {
         isSupportedSessionRequest(request) && handleSessionRequest(request)
-      const sessionDeleteListener = () => dispatch({ type: WalletConnectActionType.DELETE_SESSION })
-      const sessionUpdateListener = (session: Partial<SessionTypes.Struct>) =>
-        dispatch({ type: WalletConnectActionType.UPDATE_SESSION, payload: session })
+      }
+      const sessionDeleteListener = ({
+        topic,
+      }: SignClientTypes.EventArguments['session_delete']) => {
+        dispatch({ type: WalletConnectActionType.DELETE_SESSION, payload: { topic } })
+      }
+      const sessionUpdateListener = ({
+        topic,
+        params,
+      }: SignClientTypes.EventArguments['session_update']) => {
+        dispatch({ type: WalletConnectActionType.UPDATE_SESSION, payload: { ...params, topic } })
+      }
       const sessionPingListener = () => {}
       const pairingPingListener = () => {}
 
-      const pairingDeleteListener = () => dispatch({ type: WalletConnectActionType.DELETE_SESSION })
+      const pairingDeleteListener = ({ topic }: PairingEvent) => {
+        dispatch({ type: WalletConnectActionType.DELETE_SESSION, payload: { topic } })
+      }
 
-      const pairingExpireListener = () => dispatch({ type: WalletConnectActionType.DELETE_SESSION })
```

Comment:
really not a fan of implicit return because without a block body debugging is not possible